src/server/pkg/work/task_queue.go

```diff
@@ -0,0 +1,104 @@
+package work
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/cevaris/ordered_map"
+)
+
+const (
+	waitTime = 100 * time.Millisecond
+)
+
+type subtaskFunc func(context.Context) error
+
+// (bryce) need a remap operation for task queue because the in memory maps
+// do not free memory after deletions.
+type taskQueue struct {
+	tasks *ordered_map.OrderedMap
+	mu    sync.Mutex
+}
+
+func newTaskQueue(ctx context.Context) *taskQueue {
+	tq := &taskQueue{
+		tasks: ordered_map.NewOrderedMap(),
+	}
+	go func() {
+	NextSubtask:
+		for {
+			select {
+			case <-ctx.Done():
+				fmt.Printf("task queue context canceled: %v\n", ctx.Err())
```

Comment:
This doesn't actually do anything about it - this will loop indefinitely, printing this out.

Reply:
Fixed.